File: API/Helpers/UserParams.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace API.Helpers
{
    public class UserParams:PaginationParams
    {/* 
        private const int MaxPageSize=50;
        public int PageNumber{get;set;}=1;

        private int _pageSize=10;

        
        public int PageSize
        {
            get =>  _pageSize;
            set => _pageSize=(value>MaxPageSize)?MaxPageSize:value;
        } (L177)*/
        public string CurrentUserName { get; set; }
        public string Gender { get; set; }
        public int minAge { get; set; }=18;
        public int maxAge { get; set; }=150;
        public string OrderBy { get; set; }="lastActive";
    }
}

File: client/src/app/_models/userParams.ts
/* import { User } from "./user"; */

import { User } from "./user";

export class UserParams

{//(159)
    
    gender:string;
    /* minAge:18;
    maxAge:99;
    pageNumber:1;
    pageSize:5; */
    minAge:number;
    maxAge:number;
    pageNumber:number;
    pageSize:number;
    orderBy:string;
    constructor(user:User|null) 
    {
        this.gender=user?.gender==='female'? 'male':'female';
        this.minAge=18;
        this.maxAge=99;/*  */
        
        this.pageNumber=1;
        this.pageSize=5; 
        this.orderBy='lastActive'
    }



}